### DISCUSSION TITLE
Variant stock is not available for sales order

### DISCUSSION THREAD

**Discussion by HannesMorre** [inventree/InvenTree#5401] [Q&A]
I added a template part of which the variants are in stock to a sales order, but there it says that there is no stock available. Shouldn't the variant stock also be counted as available for a sales order?

**Comment by SchrodingersGat** [comment]
Variant stock should be available - have you checked that the variants you are interested in are labelled as "saleable"?

**Comment by HannesMorre** [comment]
I checked this but it is not working as it's supposed to I think.
This is the template part:
![canio template](https://github.com/inventree/InvenTree/assets/71696745/3270a275-e301-4643-8fd1-3cc43c532c26)
As you can see available stock is 9 and 6 are requested for a sales order. The available stock is stock from a variant part.
The screenshot below shows that this variant part is also salable
![image](https://github.com/inventree/InvenTree/assets/71696745/8c70ca2c-a7e4-4d78-95c7-e241edea95fd)
But when I look at the sales order it says that there is no stock available:
![image](https://github.com/inventree/InvenTree/assets/71696745/e8169e97-e82a-4078-93f1-4313b0f2854d)

**Comment by SchrodingersGat** [comment]
Oh, ok I misunderstood originally. You are saying that if you select a *template* part for a sales order, you should be able to select *variant* stock to allocate against the line?

**Reply by HannesMorre** [reply]
That is what I'm trying to achieve, yes. But I have a feeling, after reading your answer, it is not supposed to work like this?

**Comment by HannesMorre** [comment]
I can allocate the variant stock even though there is "no available stock".

![image](https://github.com/inventree/InvenTree/assets/71696745/27e5a473-acef-407b-aceb-0d820f2f6d11)

**Reply by SchrodingersGat** [reply]
@HannesMorre I think that this is the actual bug - the "available stock" does not include variants!

**Reply by HannesMorre** [reply]
So I should not be able to assign variant stock to a sales order?

**Reply by HannesMorre** [reply]
Is there a possibility to add substitute parts for sales orders?
What I mean is that if an article is not available in stock but a substitute part is, then this one can be assigned to the sales order instead of the original one.

**Reply by SchrodingersGat** [reply]
@HannesMorre you should be able to assign variant stock (as you said you are able to do). The issue is that the available variant stock is not included in the "count" which is displayed in the sales order table

**Comment by SchrodingersGat** [comment]
@HannesMorre this should now be fixed in https://github.com/inventree/InvenTree/pull/5415
